hypothesis-python/src/hypothesis/internal/compat.py

```diff
@@ -120,15 +121,14 @@ def get_type_hints(thing):
                         hints[p.name] = typing.Optional[p.annotation]
                     else:
                         hints[p.name] = p.annotation
-    except (AttributeError, TypeError, NameError):
+                else:  # pragma: no cover
+                    pass
```

Comment:
I tried putting `#pragma: no branch` on the `if` arm, but the autoformatter moved it onto the closing paren line and then coverage kept complaining.

So I took the extreme route of adding an explicit `else` and excluding that instead.

(This exclusion is also the one I'm least confident about, since I don't really understand this deep typing/introspection stuff in detail. But it seems *probably* fine?)

Reply:
Good motivation for testing coverage of the compat code: I should write a test that covers this branch.

(this *is* rather dark magic though, my attempt just now didn't cover it)